src/main/java/io/stargate/sgv2/jsonapi/service/updater/DocumentUpdater.java

```diff
@@ -93,9 +93,13 @@ private boolean replace(ObjectNode docToUpdate, boolean docInserted) {
     if (JsonUtil.equalsOrdered(compareDoc, replaceDocument())) {
       return false;
     }
-    // remove all data and add _id as first field
+    // remove all data and add _id as first field; either from original document or from replacement
     docToUpdate.removeAll();
-    if (idNode != null) docToUpdate.set(DocumentConstants.Fields.DOC_ID, idNode);
+    if (idNode != null) {
+      docToUpdate.set(DocumentConstants.Fields.DOC_ID, idNode);
+    } else if (replaceDocumentId != null) {
```

Comment:
Without this, `_id` from replacement document not added since it was removed from `replaceDocument()` earlier.